packages/lessons-worker/src/lib/generate-youtube-srt/main.py

```
@@ -10,10 +10,23 @@ def main():
+
     try:
         # Try to get Arabic subtitles first, fall back to auto-generated if not available
         try:
-            transcript = YouTubeTranscriptApi.get_transcript(video_id, languages=['ar'])
+            ytt_api = YouTubeTranscriptApi()
+            print("Trying to get Arabic subtitles...")
+            fetched_transcript = ytt_api.fetch(video_id, languages=['ar'])
+            print(f"Fetched {len(fetched_transcript)} subtitles")
+            # Save as SRT
+            with open(f"{output_dir}/{video_id}.srt", "w", encoding="utf-8") as srt_file:
+                for i, entry in enumerate(fetched_transcript, 1):
```

Ruff (0.12.2)

Loop control variable `i` not used within loop body

(B007)